source/indexer/contracts/Indexer.sol

```diff
@@ -277,6 +272,50 @@ contract Indexer is IIndexer, Ownable {
     return indexes[signerToken][senderToken].getLocators(startAddress, count);
   }
 
+  /**
+    * @notice Gets the Stake Amount for a User
+    * @param user address User who staked
+    * @param signerToken address Signer token the user staked on
+    * @param senderToken address Sender token the user staked on
+    * @return uint256 Amount the user staked
+    */
+  function getStakedAmount(
+    address user,
+    address signerToken,
+    address senderToken
+  ) public view indexExists(signerToken, senderToken) returns (uint256) {
+
+    // Return the score, equivalent to the stake amount.
+    return indexes[signerToken][senderToken].getScore(user);
+  }
+
+  function _updateIntent(
+    address user,
+    address signerToken,
+    address senderToken,
+    uint256 newAmount,
+    bytes32 newLocator,
+    uint256 oldAmount
+  ) internal {
+    // If the new stake is bigger, collect the difference.
+    if (oldAmount < newAmount) {
+      require(stakingToken.transferFrom(user, address(this), newAmount - oldAmount),
+        "UNABLE_TO_STAKE");
+    }
+
+    // If the old stake is bigger, return the excess.
+    if (newAmount < oldAmount) {
```

Comment:
note that safemath not needed to be used since overflow is done by the less than check